app/views/handlers/list_remove_question.py

```diff
@@ -20,10 +20,9 @@ def is_location_valid(self):
         return True
 
     def handle_post(self):
-        if (
-            self.form.data[self.parent_block["remove_answer"]["id"]]
-            == self.parent_block["remove_answer"]["value"]
-        ):
+        answer_action = self._get_answer_action()
+
+        if answer_action and answer_action["type"] == "RemoveAnswerForListItem":
```

Comment:
Interesting point, however with your above suggestion about RedirectToAddBlock, this should be RedirectToRemoveBlock

Reply:
I don't think it should. This action removes all answers related to the list item id which the remove is called on.

Reply:
@LJBabbage I think this may have accidentally been marked as resolved. The `RedirectToRemoveBlock` action name is wrong, it should be something like `RemoveListItemAndAnswers` or `RemoveAnswersForListItem`.

Reply:
Darn you are right, I was thinking it was redirecting to the remove block, its not, its already in there

Reply:
I am not against it but runner internally calls this `remove_list_item_and_answers` because it is removing both the list item from the list store and its answers and not just the answers for the list item, therefore, `RemoveListItemAndAnswers` is a better representation of the actual behaviour of the action. Thought? @LJBabbage @ajmaddaford